### ISSUE TITLE
Change SecretStoreSpec.RefreshInterval int -> *metav1.Duration 

### ISSUE THREAD

**Issue by zhilyaev** [external-secrets/external-secrets#2977]
**Is your feature request related to a problem? Please describe.**
Change RefreshInterval type to *metav1.Duration. Cause All RefreshInterval has the same type.

**Describe the solution you'd like**
I wan't use 1h, 1m with SecretStoreSpec.RefreshInterval

```golang
// SecretStoreSpec defines the desired state of SecretStore.
type SecretStoreSpec struct {
	// Used to select the correct ESO controller (think: ingress.ingressClassName)
	// The ESO controller is instantiated with a specific controller name and filters ES based on this property
	// +optional
	Controller string `json:"controller,omitempty"`

	// Used to configure the provider. Only one provider may be set
	Provider *SecretStoreProvider `json:"provider"`

	// Used to configure http retries if failed
	// +optional
	RetrySettings *SecretStoreRetrySettings `json:"retrySettings,omitempty"`

	// Used to configure store refresh interval in seconds. Empty or 0 will default to the controller config.
	// +optional
	RefreshInterval *metav1.Duration `json:"refreshInterval,omitempty"`

	// Used to constraint a ClusterSecretStore to specific namespaces. Relevant only to ClusterSecretStore
	// +optional
	Conditions []ClusterSecretStoreCondition `json:"conditions,omitempty"`
}
```

**Comment by shuheiktgw** [comment]
This makes sense. We could probably change that when releasing v1.

**Comment by shuheiktgw** [comment]
This somewhat relates to https://github.com/external-secrets/external-secrets/issues/2761

**Comment by Skaronator** [comment]
Just run into the same issue. Is it possible to implement this (somewhat breaking?) change in the existing `SecretStore/v1` Resource or do we need a v2?

**Comment by Skarlso** [comment]
Yah, since this is a type change, it would need to be a v2. :/

**Comment by evrardj-roche** [comment]
I removed the milestone, it was incorrect (v1). I think we should keep have another milestone to track our big "next gen" improvement. @Skarlso can you help there, by creating a new milestone to track this? (let's not call it v2, it will confuse people I think...).

**Comment by Skarlso** [comment]
Hm, calling it anything else sort of falls flat when trying to estimate / track whatever it's doing or supposed to achieve. Next-gen tells me nothing. :) We either need a version or something that we want to achieve.

**Comment by evrardj-roche** [comment]
I can agree with that. At the same time, if we follow semver, that version might have to be moved over time. I would follow your lead there.

**Comment by Skarlso** [comment]
argh, naming is so hard. :D I don't knoooowww. :D

**Comment by Skarlso** [comment]
Let's call it something like `2.*`?

**Comment by evrardjp** [comment]
We can also ask @gusfcarvalho or @moolen, as they have been doing most of the new stuff. I don't care about the name, I only care about the fact we are _all talking the same language_ . In this instance, I am afraid we target a milestone `2.X`, then ppl (who would be less familiar with the project) would see the milestone and be confused "but there has been a 3.0 release delivered, why is this not in ?!?"

**Comment by alekc** [comment]
We can change the type without the breaking change though?

Couple of options here: 
1) change the type to refreshInterval: *metav1.Duration and a conversion webhook that maps v1 int seconds ⇄ new Duration. The conversion is lossless and unambiguous: 300 ⇄ "5m0s", pure arithmetic. That part is trivial. The expensive part: ESO has never built a conversion webhook, so you'd stand up Hub/Convertible (ConvertTo/ConvertFrom) for the entire SecretStore + ClusterSecretStore types (the whole provider spec, every backend), plus conversion.strategy: Webhook + caBundle + RBAC. Disproportionate for one int field on its own but might be worth having this as a scaffolding in place for the future. 
2)  Make refreshInterval accept both int and string via a custom int-or-string Duration type (x-kubernetes-int-or-string). Same field, same version, 300 and 1h both parse until v2. Non-breaking and one field, but non-idiomatic: it disables structural-schema validation niceties, confuses kubectl explain, needs a hand-written UnmarshalJSON

**Comment by gusfcarvalho** [comment]
Ever since 1.0 we stopped shipping conversion webhooks… they’re a pain to maintain 

2 for non breaking would make more sense

**Comment by evrardj-roche** [comment]
I agree with @gusfcarvalho : Bundling breaking changes has benefits. Other projects have done it, so we would not be the only ones.

**Comment by alekc** [comment]
you can assign this one to me then, I will draft up the 2 variant

**Comment by evrardj-roche** [comment]
cool thanks!

**Comment by alekc** [comment]
Done: https://github.com/external-secrets/external-secrets/pull/6594

Should accept now either string or int. left a comment stating that for v2 we need to migrate to a proper duration. @gusfcarvalho / @evrardj-roche could you have a look?

**Comment by evrardj-roche** [comment]
Ah I see where are you heading. I will have a looksie at your PR today! :) 

Thank you for the work 🙇
